modules/core/src/main/java/org/apache/ignite/cache/QueryEntity.java

```diff
@@ -669,6 +687,84 @@ public QueryEntity addQueryField(String fullName, String type, String alias) {
         return this;
     }
 
+    /**
+     * Returns INLINE_SIZE for PK index.
+     *
+     * @return INLINE_SIZE for PK index.
+     */
+    public Integer getPrimaryKeyInlineSize() {
+        return pkInlineSize;
+    }
+
+    /**
+     * Sets INLINE_SIZE for PK index.
+     *
+     * @param pkInlineSize INLINE_SIZE for PK index, when {@code null} - inline size is calculated automativally.
+     * @return {@code this} for chaining.
+     */
+    public QueryEntity setPrimaryKeyInlineSize(Integer pkInlineSize) {
+        if (pkInlineSize != null && pkInlineSize < 0) {
+            throw new CacheException("Inline size for sorted primary key cannot be negative. "
+                    + "[inlineSize=" + pkInlineSize + ']');
+        }
+
+        this.pkInlineSize = pkInlineSize;
+
+        return this;
+    }
+
+    /**
+     * Returns INLINE_SIZE for affinity field index.
+     *
+     * @return INLINE_SIZE for affinity field index.
+     */
+    public Integer getAffinityFieldInlineSize() {
+        return affFieldInlineSize;
+    }
+
+    /**
+     * Sets INLINE_SIZE for AFFINITY_KEY index.
+     *
+     * @param affFieldInlineSize INLINE_SIZE for AFFINITY_KEY index, when {@code null} - inline size is calculated automativally.
+     * @return {@code this} for chaining.
+     */
+    public QueryEntity setAffinityKeyInlineSize(Integer affFieldInlineSize) {
+        if (affFieldInlineSize != null && affFieldInlineSize < 0) {
+            throw new CacheException("Inline size for affinity filed index cannot be negative. "
+                    + "[inlineSize=" + affFieldInlineSize + ']');
+        }
+
+        this.affFieldInlineSize = affFieldInlineSize;
+
+        return this;
+    }
+
+    /**
+     * The property is used for sorted index and composite primary key.
+     *
+     * @return  {@code true} if the PK index is created on fields of PK.
+     * {@code false} in case the PK index is created on the whole key (composite binary object).
+     * {@code null} - compatible behavior (unwrap for a table created by SQL and wrapped key for a table created by API).
+     */
+    public Boolean getUnwrapPrimaryKeyFieldsForSortedIndex() {
```

Comment:
Do we have non-sorted primary key index?

Reply:
I guess yes. Real _KEY object. I guess it's a matter of terminology: key / value vs table behavior.